src/components/task/uc/deleteUserData.uc.unit.test.js

```diff
@@ -0,0 +1,332 @@
+const sinon = require('sinon');
+const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
+const { ObjectId } = require('mongoose').Types;
+
+// const chaiAsPromised = require('chai-as-promised');
+const { deleteUserRelatedData } = require('./deleteUserData.uc');
+const repo = require('../repo/task.repo');
+const { toString: idToString } = require('../../../helper/compare').ObjectId;
+// const testObjects = require('../../../../test/services/helpers/testObjects');
+
+const { expect } = chai;
+const mapByName = (deleteUserDataFunctions) =>
+	deleteUserDataFunctions.reduce((map, fun) => {
+		map[fun.name] = fun;
+		return map;
+	}, {});
+
+chai.use(chaiAsPromised);
+
+const getExpectedUpdateMany = (modifiedDocuments) => ({ success: true, modifiedDocuments });
+const getExpectedDeleteMany = (deletedDocuments) => ({ success: true, deletedDocuments });
+
+describe('in "deleteUserData.uc" the function', () => {
+	describe('deleteUserData', () => {
+		const userId = new ObjectId();
+		const replaceUserId = new ObjectId();
+
+		// true result
+		it('should return a function that return an array.', async () => {
+			const result = deleteUserRelatedData();
+			expect(result).to.be.an('array').with.lengthOf(5);
+			expect(result[0], 'where first element is a function').to.be.a('function');
+			expect(result[1], 'where second element is a function').to.be.a('function');
+			expect(result[2], 'where third element is a function').to.be.a('function');
+			expect(result[3], 'where fourth element is a function').to.be.a('function');
+			expect(result[4], 'where fourth element is a function').to.be.a('function');
+		});
+
+		describe('deletePrivateSubmissions', () => {
+			const initStubs = ({ studentId }) => {
```

Comment:
This could also be moved in a `beforeEach` inside the describe. But this is also fine for me.